channel/buffs.go

```diff
@@ -165,8 +165,8 @@ type CharacterBuffs struct {
 	expireAt          map[int32]int64  // sourceID -> unix ms expiry
 }
 
-func NewCharacterBuffs(p *Player) *CharacterBuffs {
-	return &CharacterBuffs{
```

Comment:
Is this the root cause of the pointer going stale? If so I am not sure why it would be the case.